src/redux/selectors.js

```diff
@@ -17,11 +35,27 @@ const loadingSelector = state => state.ui.loading;
 const projectSelector = id => state =>
   state.entities.projects.find(p => p.id === id);
 
-const reportSelector = state => ({
-  sprints: state.entities.sprints,
-  stories: state.entities.stories,
-  timeEntries: state.entities.timeEntries,
-});
+const reportSelector = state => {
+  let stories = state.entities.stories;
+
+  if (
+    !isEmpty(state.entities.timeEntries) &&
+    !isEmpty(state.entities.stories)
+  ) {
+    const storyTimeEntries = getStoryTimeEntries(state.entities.timeEntries);
+
+    stories = state.entities.stories.map(story => ({
```

Comment:
🤓 This is silly and definitely premature optimization, but it occurred to me just now that we could refactor this to avoid the `O(s * t)` we're getting by scanning all of `timeEntries` for each `story`. 

```
const hoursByStoryId = timeEntries.reduce(() => {
  // TODO: Accumulate the sum of hours per storyId based on the entry.
}, {});

stories = state.entities.story.map(
  story => ({ ...story, trackedHours: get(hoursByStoryId, story.id, 0) },
); 
```